### PR TITLE
Feat/net097 29/unit crud

### PR DESCRIPTION
**Pull request by Alex-knu**
Task 29(https://military-manager.atlassian.net/browse/NET097-6?atlOrigin=eyJpIjoiYzAzNGM0MTAzZmNiNGE1ZThhNzZkZjMyYzFlNWYyNjYiLCJwIjoiaiJ9)

### CODE REVIEW

File: MilitaryManager.Core/Entities/UnitEntity/Unit.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public class Unit : IBaseEntity<int>
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public Unit Parent { get; set; }
+
+        public ICollection<Unit> SubUnits { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/UnitEntity/Unit.cs:15]: Should be discussed

File: MilitaryManager.Units.API/appsettings.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=BAGUTA;Initial Catalog=Units;User ID=sa;Password=password1;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False"
+    "DefaultConnection": "Data Source=DESKTOP-4TBBV5N;Initial Catalog=Units;User ID=Gwa_System;Password=T&t137S5tP!+3SMJQ<r0KJ;Connect Timeout=10;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False"
```
Comment (YevheniiVovk) [inline MilitaryManager.Units.API/appsettings.json]: Please DO NOT commit any configuration changes

File: docker-compose.override.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ services:
     environment:
       - ASPNETCORE_ENVIRONMENT=Development
       - ASPNETCORE_URLS=https://+:443;http://+:80
-      - ConnectionStrings__DefaultConnection=Server=host.docker.internal;Database=Units;User ID=sa;Password=password1;MultipleActiveResultSets=true;Application Name=MilitaryManager.Units.API
+      - ConnectionStrings__DefaultConnection=Server=host.docker.internal;Database=Units;User ID=Gwa_System;Password=T&t137S5tP!+3SMJQ<r0KJ;MultipleActiveResultSets=true;Application Name=MilitaryManager.Units.API
```
Comment (YevheniiVovk) [inline docker-compose.override.yml]: Please DO NOT commit any configuration changes

File: MilitaryManager.Units.API/Controllers/UnitController.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        public UnitController(IUnitService unitServices)
+        {
+            _unitServices = unitServices;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetTree()
+        {
+            return Ok(await _unitServices.GetUnitsTreeAsync());
+        }
```
Comment (YevheniiVovk) [inline MilitaryManager.Units.API/Controllers/UnitController.cs]: REST miss

File: MilitaryManager.Core/Services/UnitService.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            await _unitRepository.AddAsync(unit);
+            await _unitRepository.SaveChangesAcync();
+        }
+
+        public async Task UpdateUnitAsync(UnitDTO query)
+        {
+            var unit = _mapper.Map<Unit>(query);
+            await _unitRepository.UpdateAsync(unit);
+            await _unitRepository.SaveChangesAcync();
+        }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Services/UnitService.cs]: Lets discuss void types

File: MilitaryManager.Core/Interfaces/Services/IUnitService.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+using System.Threading.Tasks;
+
+namespace MilitaryManager.Core.Interfaces.Services
+{
+    public interface IUnitService
+    {
+        Task<IEnumerable<UnitDTO>> GetUnitsTreeAsync();
+        Task CreateUnitAsync(UnitDTO query);
+        Task UpdateUnitAsync(UnitDTO query);
+        Task DeleteUnitAsync(UnitDTO query);
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Interfaces/Services/IUnitService.cs]: return type should be discussed

File: MilitaryManager.Core/DTO/Units/UnitDTO.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    public class UnitDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public UnitDTO Parent { get; set; }
+
+        public ICollection<UnitDTO> SubUnits { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/DTO/Units/UnitDTO.cs]: Lets use list here or IEnumerable but do not use ICollection

File: MilitaryManager.Core/Services/UnitService.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<UnitDTO>> GetUnitsTreeAsync()
+        {
+            var specificationUnits = new Units.UnitsList();
+            var unitList = await _unitRepository.GetListBySpecAsync(specificationUnits);
+            var unitsTree = unitList.Where(x => x.ParentId == null);
+
+            return _mapper.Map<IEnumerable<UnitDTO>>(unitsTree);
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Services/UnitService.cs]: Should be rewrited

File: MilitaryManager.Core/Helpers/ApplicationProfile.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+using MilitaryManager.Core.DTO.Units;
+using MilitaryManager.Core.Entities.UnitEntity;
+
+namespace MilitaryManager.Core.Helpers
+{
+    public class ApplicationProfile : Profile
+    {
+        public ApplicationProfile()
+        {
+            CreateMap<UnitDTO, Unit>();
```
Comment (vladtymo) [inline MilitaryManager.Core/Helpers/ApplicationProfile.cs]: You can use ReverseMap() method

**Review by vladtymo — APPROVED** [review]

**Review by vladtymo — APPROVED** [review]

**Review by NataliaHerus — APPROVED** [review]

**Pull request merged**